File: C/Examples/bubble_sort_0001.c
/*************************************************************************
    > File Name: bubble_sort_0001.c
    > Author: lwei2
    > Mail: lwei2@IT.com 
    > The Description :
 ************************************************************************/

#include <stdio.h>

int main(void)
{
	int a[] = {
		800,2,3,-58,34,76,32,43,56,-70,35,-234,532,543,2500
	};
	int tmp;
	int len = sizeof(a)/sizeof(a[0]);
	for(int i = 0; i < len - 1; ++i)
	{
		for(int j = 0; j < len - 1 - i; ++j)
		{
			if(a[j] > a[j+1])
			{
				tmp = a[j];
				a[j] = a[j+1];
				a[j+1] = tmp;
			}
		}
	}

	for(int i = 0; i < len; ++i)
		printf("%d\x20",a[i]);
	printf("\n");
	return 0;
}

File: TheDataStructure/Algorithm/SortAlgorithm/BubbleSort/Java/BubbleSort_0001.java
/*************************************************************************
    > File Name: BubbleSort_0001.java
    > Author: lwei2
    > Mail: lwei2@IT.com 
    > The Description :
 ************************************************************************/
public static void BubbleSort(int arr[])
{
	for(int i = 1; i < arr.length; i++)
	{
		for(int j = 0; j < arr.length - i; j++)
		{

			if(arr[j] > arr[j+1])
			{
				int tmp = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = tmp;
			}
		}
	}
}
